compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -974,36 +971,69 @@ LogicalResult DataTiledMMAAttr::populateOperandOffsetsSizesStrides(
     DBGS() << "    swizzle: " << swizzle << "\n";
   });
 
-  // Populate tile sizes.
   MLIRContext *ctx = builder.getContext();
   SmallVector<OpFoldResult> tileSizes = getAsIndexOpFoldResult(
       ctx, sliceSwizzledShape(swizzle, [](TileSwizzle::Dim d) {
         return d.kind != TileSwizzle::Dim::Kind::CrossThread;
       }));
 
-  // Populate tile offsets by delinearizing threadId over the CrossThread dims.
-  // Since the AffineDelinearizeIndexOp does not bound the input index, we
-  // must bound the threadId by the product of the offset ranges.
-  SmallVector<int64_t> tileOffsetsBasis =
+  // Most of the rest of this function is the computation of the offsets.
+  // The basic idea is to delinearize the threadId over the basis of
+  // cross-thread dimensions. The main subtlety is that the TileSwizzles refer
+  // only to the layout dims, and do not reflect a possible additional
+  // thread-distribution-only dimension present on some architectures (RDNA3).
+  // When such an extra dim exists, multiple threads are reading the same data.
+  // So we need to distinguish layoutThreadSizes vs. distributionThreadSizes.
```

Comment:
So if we have an intrinsic that takes let's say 32 elements, we are essentially broadcasting it to size 64, and distributing it. And when doing anything else, we ignore the broadcasted dimension, to get the same data on different threads.

Reply:
Yes.